Source/WebCore/html/FormController.cpp

```diff
@@ -244,27 +241,23 @@ static String formStateSignature()
 Vector<AtomString> FormController::formElementsState(const Document& document) const
 {
     struct Control {
-        Ref<const HTMLFormControlElementWithState> control;
+        Ref<const ValidatedFormListedElement> control;
         String formKey;
     };
 
     Vector<Control> controls;
     {
         // FIXME: We should be saving the state of form controls in shadow trees, too.
         FormKeyGenerator keyGenerator;
-        for (auto& control : descendantsOfType<HTMLFormControlElementWithState>(document)) {
-            ASSERT(control.insertionIndex());
-            if (control.shouldSaveAndRestoreFormControlState())
-                controls.append({ control, keyGenerator.formKey(control) });
+        for (auto& element : descendantsOfType<Element>(document)) {
+            if (auto* control = const_cast<Element&>(element).asValidatedFormListedElement()) {
+                if (control->isCandidateForSavingAndRestoringState())
+                    controls.append({ *control, keyGenerator.formKey(*control) });
+            }
         }
     }
     if (controls.isEmpty())
         return { };
-    std::sort(controls.begin(), controls.end(), [](auto& a, auto& b) {
```

Comment:
Why are we deleting this code to sort controls by form keys?

Reply:
Fixed in [2e90026](https://github.com/WebKit/WebKit/pull/4988/commits/2e90026b9f7754f3667525f07409e922be9ed92e): I was meaning to remove `insertionIndex()` thing only as it would inflate `sizeof` of some built-in controls. The updated implementation of `formElementsState()` is tested to be functionally equivalent to the `std::sort` one.